tapa/graphir/types/expressions/expression.py

```diff
@@ -347,3 +347,69 @@ def rewrite(self, idmap: dict[str, "Expression"]) -> "Expression":
                 new_expr.append(part)
 
         return Expression(tuple(new_expr))
+
+    @staticmethod
```

Comment:
Prefer `@classmethod`? It is generally preferred for factory/constructor functions. `@staticmethod` is generally not preferred (free functions are usually more appropriate).